### PR TITLE
Refactor GitHub Actions structure

### PR DESCRIPTION
**Pull request by slvrtrn**
Resolves https://github.com/ClickHouse/clickhouse-js/issues/3

### CODE REVIEW

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -47,11 +45,105 @@ jobs:
...
+        run: |
+          npm run test:unit
+
+  integration-tests-local-single-node:
+    needs: build
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        node: [14, 16]
```
Comment (mshustov) [inline .github/workflows/run-tests.yml:63]: Optional: maybe we can use [yaml alias and anchor](https://yaml.org/spec/1.2-old/spec.html#id2765878) mechanism to reference duplicate declarations?

File: __tests__/integration/abort_request.test.ts [hunk 2]
```diff
@@ -128,19 +126,23 @@ describe('abort request', () => {
         });
     });
 
-    it('ClickHouse server must cancel query on abort', async () => {
+    // FIXME: it does not work with ClickHouse Cloud.
+    //  Active queries never contain the long running query unlike local setup.
+    it.skip('ClickHouse server must cancel query on abort', async () => {
```
Comment (mshustov) [inline __tests__/integration/abort_request.test.ts:131]: Let's add a way to skip some tests based on env type. can be done as a follow-up.

File: __tests__/integration/command.test.ts [hunk 3]
```diff
@@ -42,11 +41,11 @@ describe('command', () => {
     expect(table.create_table_query).to.be.a('string');
   });
 
-  it('does not swallow ClickHouse error', (done) => {
+  it.skip('does not swallow ClickHouse error', (done) => {
```
Comment (mshustov) [inline __tests__/integration/command.test.ts]: Does it fail on Cloud? If so, let's add a comment explaining why it's skipped.

File: .github/workflows/run-tests.yml (hunk 1 above)
Comment (slvrtrn) [reply .github/workflows/run-tests.yml:63]: will do as a follow-up

File: __tests__/integration/command.test.ts (hunk 3 above)
Comment (slvrtrn) [reply __tests__/integration/command.test.ts]: Will do a fix based on [this issue](https://github.com/ClickHouse/ClickHouse/issues/15124#issuecomment-889005203)

**Pull request merged**